### PR TITLE
Support version endpoints for pods.

### PR DESCRIPTION
**Pull request by jasongilanfarr**
Summary: - Add support for the version endpoints described in DCOS-1004

Test Plan: sbt test

Reviewers: jdef, aquamatthias

Subscribers: marathon-team

Differential Revision: https://phabricator.mesosphere.com/D27

### CODE REVIEW

File: src/main/scala/mesosphere/marathon/api/v2/PodsResource.scala [hunk 1]
```diff
@@ -198,6 +198,34 @@ class PodsResource @Inject() (
...
+  @GET
+  @Timed
+  @Path("""{id:+}::versions/{version}""")
+  def version(@PathParam("id") id: String, @PathParam("version") versionString: String,
+             @Context req: HttpServletRequest): Response = authenticated(req) { implicit identity =>
+    import PathId._
+    val version = Timestamp(versionString)
+    withValid(id.toRootPath) { id =>
+      result(podSystem.version(id, version)).fold(notFound(id)) { pod =>
+        ok(Json.stringify(Json.toJson(Raml.toRaml(pod))))
```
Comment (unterstein) [inline src/main/scala/mesosphere/marathon/api/v2/PodsResource.scala]: maybe use helper functionality ```ok(marshal(pod))```?

File: src/test/scala/mesosphere/marathon/api/v2/PodsResourceTest.scala [hunk 2]
```diff
@@ -4,116 +4,191 @@ import javax.servlet.http.HttpServletResponse
...
-      body should not be (None)
-      body.foreach(_ should include("mypod does not exist"))
+      withClue(s"response body: ${response.getEntity}") {
+        response.getStatus should be(HttpServletResponse.SC_NOT_FOUND)
+        val body = Option(response.getEntity.asInstanceOf[String])
+        body should not be (None)
+        body.foreach(_ should include("mypod does not exist"))
+      }
+    }
+
```
Comment (unterstein) [inline src/test/scala/mesosphere/marathon/api/v2/PodsResourceTest.scala]: all changes in ```PodsResourceTest``` before this line was just changing the test format style, or did I miss a semantically change?

**Comment by jdef** [comment]
commented on D27: seems to be missing authz checks

**Comment by unterstein** [comment]
@jdef No endpoint (except status(id: String) uses authz in PodsResource. Should this be corrected for all the other endpoints as well?

**Comment by jasongilanfarr** [comment]
That's why I didn't do it (didn't even notice that status did have it)- let
address this separately.

**Comment by jdef** [comment]
@jasongilanfarr @unterstein every other endpoint uses authz (withAuthorization)

**Comment by jasongilanfarr** [comment]
Oh- I'm mistaken hard to see in there.

**Comment by jasongilanfarr** [comment]
merged into pods

**Pull request closed without merging**